sentry.cpp

```diff
@@ -102,12 +203,29 @@ void serialize(const sentry_event_t *event)
 
 int sentry_init(const sentry_options_t *options)
 {
-    auto err = init(options);
+    if (options->dsn == nullptr)
+    {
+        if (options->debug)
+        {
+            fprintf(stderr, "Not DSN specified. Sentry SDK will be disabled.\n");
+        }
+        return SENTRY_ERROR_NO_DSN;
+    }
+
+    std::string minidump_url;
+    auto err = minidump_url_from_dsn(strdup(options->dsn), minidump_url);
```

Comment:
strdup will leak here. After the parsed dsn is not needed any more after the call you need to `free()` it.